block/block_sync.go

```diff
@@ -113,7 +114,13 @@ func (bSyncService *BlockSyncService) WriteToBlockStoreAndBroadcast(ctx context.
 
 	// Broadcast for subscribers
 	if err := bSyncService.sub.Broadcast(ctx, block); err != nil {
-		bSyncService.logger.Error("failed to broadcast block", "error", err)
+		// for the genesis header, broadcast error is expected as we have already initialized the store
+		// for starting the syncer. Hence, we ignore the error.
+		// exact reason: validation failed, err header verification failed: known header: '1' <= current '1'
```

Comment:
if the error is expected for genesis, should we just not call broadcast? 

Or can we check for that error explicitly (the one listed in the comment) to avoid missing other potential errors returned from broadcast during genesis?

Reply:
broadcast is needed in the sequencer to share the produced block with p2p peers. ideally, we should not init and start the syncer and then broadcast which results in double init of the genesis block. we should only do the broadcast and trigger syncer to init after that. i have created an issue to tackle this in v1: https://github.com/rollkit/rollkit/issues/1501

with regards to catching the explicit error, unfortunately, only `validation failed` is thrown without the actual error details, hence we cannot match the exact/partial error. This is something go-header can fix and i have created an issue for it: https://github.com/celestiaorg/go-header/issues/146

Reply:
https://github.com/celestiaorg/go-header/issues/146#issuecomment-1905191830